portfolio/src/main/webapp/script.js

```diff
@@ -47,3 +47,11 @@ function addRandomFact() {
   factContainer.innerText = fact;
 }
 
+/**
+ * Fetches message from server and adds it to DOM
+ */
+function getMessage() {
+  fetch('/data').then(response => response.text()).then((message) => {
+    document.getElementById('message-container').innerHTML = message;
```

Comment:
Most of the time you don't want to assign to innerHTML. If this message is always written by you it might be fine. But if users can supply this text (like through a commenting system) it would introduce a dangerous vulnerability. Please read this section on MDN: https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#Security_considerations

Reply:
I'll keep that in mind going forward. In this case, the message from the starter code included HTML tags. Here's the line in DataServlet.java:
~~~~
response.getWriter().println("<h1>Hello Kaye!</h1>");
~~~~
Using innerText or textContent includes the tags when it's displayed on the website. Should I leave my code as is or take out the h1 tags in the java file? Or is there some alternative I can use instead?

Reply:
I see, thanks. In that case it's fine to leave as is, but please leave a JS comment saying that this is only safe because it comes from a static string in the server.